### PR TITLE
Resolve vars-on-top, array-callback-return, no-restricted-syntax, no-undef violations

### PR DESCRIPTION
**Pull request by ajpal**
@thallada @dsjen

### CODE REVIEW

File: analytics_dashboard/static/js/config.js [hunk 1]
```diff
@@ -63,11 +63,13 @@ require.config({
         backbone: {
             deps: ['underscore', 'jquery'],
             exports: 'Backbone',
+            /* eslint-disable no-undef */
             init: function(_, $) {
                 'use strict';
                 Backbone.$ = $;
                 return Backbone;
             }
+            /* eslint-enable no-undef */
```
Comment (ajpal) [inline analytics_dashboard/static/js/config.js:72]: I'm not sure if disabling the rule here is okay. It's saying Backbone is undefined, and I wasn't sure how to resolve it.

File: analytics_dashboard/static/js/test/spec-runner.js [hunk 2]
```diff
@@ -58,7 +58,7 @@
         config.callback = window.__karma__.start; // eslint-disable-line no-underscore-dangle
     }
 
-    requirejs.config(config);
+    requirejs.config(config); // eslint-disable-line no-undef
```
Comment (ajpal) [inline analytics_dashboard/static/js/test/spec-runner.js]: Same here. Should I add requirejs to the globals in the config, or disable just this line? Or is there a way to actually resolve the error?

Comment (thallada) [inline analytics_dashboard/static/js/test/spec-runner.js]: Did you try making a [`/*global ...*/` comment](http://eslint.org/docs/rules/no-undef#rule-details)? That looks like the only way to let eslint know it's a global variable.

File: analytics_dashboard/static/js/views/bar-view.js [hunk 3]
```diff
@@ -39,14 +39,12 @@ define(['d3', 'nvd3', 'underscore', 'utils/utils', 'views/chart-view'],
              * Returns function for displaying a truncated label.
              */
             truncateXTick: function(d) {
-                var self = this;
-                d = self.formatXValue(d);
-
-                var barWidth = d3.select(self.options.barSelector).attr('width'),  // jshint ignore:line
+                var self = this,
+                    formattedLabel = self.formatXValue(d),
+                    barWidth = d3.select(self.options.barSelector).attr('width'),  // jshint ignore:line
```
Comment (thallada) [inline analytics_dashboard/static/js/views/bar-view.js]: We should get rid of or replace the jshint ignore comment.

**Comment by thallada** [comment]
Awesome :+1:

**Pull request merged**